cmd/dependabot/internal/cmd/update.go

```diff
@@ -145,20 +80,117 @@ var updateCmd = &cobra.Command{
 	},
 }
 
-func readInputFile(file string) (*model.Input, []byte, error) {
+func extractInput(cmd *cobra.Command) (*model.Input, error) {
+	hasFile := file != ""
+	hasArguments := len(cmd.Flags().Args()) > 0
+	hasServer := inputServerPort != 0
+	hasStdin := doesStdinHaveData()
+
+	var count int
+	for _, b := range []bool{hasFile, hasArguments, hasServer, hasStdin} {
+		if b {
+			count++
+		}
+	}
+	if count > 1 {
+		return nil, errors.New("can only use one of: input file, arguments, server, or stdin")
+	}
+
+	if hasFile {
+		return readInputFile(file)
+	}
+
+	if hasArguments {
+		input, err := readArguments(cmd)
+		return input, err
+	}
+
+	if hasServer {
+		return server.Input(inputServerPort), nil
```

Comment:
For consistency with the rest of these and future flexibility do you want to make `server.Input()` return an `err`?

Reply:
Good suggestion, done!